cocotb/share/lib/simulator/simulatormodule.c

```diff
@@ -394,9 +399,15 @@ static PyObject *register_timed_callback(PyObject *self, PyObject *args)
         return NULL;
     }
 
-    // Extract the time
-    PyObject *pTime = PyTuple_GetItem(args, 0);
-    time_ps = PyLong_AsLongLong(pTime);
+    {   // Extract the time
+        PyObject *pTime = PyTuple_GetItem(args, 0);
+        long long pTime_as_longlong = PyLong_AsLongLong(pTime);
+        if (pTime_as_longlong < 0) {
```

Comment:
Please use the logic from https://github.com/cocotb/cocotb/pull/1137#discussion_r348406932, this doesn't handle errors correctly.